src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1055,6 +1055,26 @@ void InterpreterCreateQuery::assertOrSetUUID(ASTCreateQuery & create, const Data
     bool from_path = create.attach_from_path.has_value();
     bool is_on_cluster = getContext()->getClientInfo().query_kind == ClientInfo::QueryKind::SECONDARY_QUERY;
 
+    if (database->getEngineName() == "Replicated" && create.uuid != UUIDHelpers::Nil && !is_replicated_database_internal)
+    {
+        if (getContext()->getSettingsRef().database_replicated_allow_explicit_uuid == 0)
```

Comment:
An enum would be nice.

Reply:
This setting is not supposed to be usable :)

Reply:
Oh ok nvm.